src/Microsoft.Diagnostics.Monitoring.RestServer/Models/EventPipeProviderModel.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer.Models
+{
+    [DataContract(Name = "EventPipeProvider")]
+    public class EventPipeProviderModel
+    {
+        [DataMember(Name = "name", IsRequired = true)]
+        public string Name { get; set; }
+
+        [DataMember(Name = "keywords", IsRequired = true)]
+        public string Keywords { get; set; }
```

Comment:
I was thinking of changing the Keywords property to be long and do some kind of conversion during model binding so that if the caller posts a property value that is a string that roughly represents an integer (either base 10 or a base 16 that starts with 0x, in order to differentiate), then it can be converted to a long before the resolved EventPipeProvider is validated and passed into the controller method. This allows the conversion to participating in the nice model reporting when model binding or validation fails.

I found that you cannot use a TypeConverter on a property, as that will be ignored by deserialization. Using JsonConverter assumes that the posted data is JSON, brings in Newtonsoft.Json by default since we are compiling for netstandard2.0 (which might not be desirable if hosted by something that wants to use System.Text.Json, and I'm not certain how to provide custom errors from that (throwing certain exceptions either get swallowed or replaced by a generic error).

Anyone have suggestions?

Reply:
I can use JsonConverter to do the conversion. I found that if I throw JsonSerializationException, then I can have it report whatever message I provide that exception. If we want to support XML, then we can implement custom serialization for that too.

If I use JsonConverter, I think that we might need to provide some kind of shim for the data model and detect whether Newtonsoft.Json or System.Text.Json is being used to determine which variant of the data model to use.

Am I overthinking this?

Reply:
If this fails in the 3.1 runtime (due to missing Newtonsoft), I would just consider making it simpler and changing the data model to be a string and doing parsing afterwards. Note that for library scenarios, if we are the ones that trigger a newtonsoft load (when one is not required), that might also not be ideal.

Reply:
I can change it back and put a regex on the property so that we get the correctly specified value when it passes model validation.

Reply:
I'm trying to make use of model validation as much as possible because it gives a decent experience for determining where the issue exists in the model that is passed in. For example, if I try to pass a configuration where the first provider has an invalid keyword, it'll return a result that looks like:
```
{
    "configuration": [
        "The configuration field is required."
    ],
    "providers[0].keywords": [
        "The value '1A' is not a valid integer."
    ]
}
```